drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1146 @@
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the writing procedure.
+*******************************************************************************/
+static int adxl313_iio_write_calibbias(void *dev, char *buf, uint32_t len,
+				       const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t ret;
+	int32_t val;
+	enum adxl313_op_mode op_mode;
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+
+
+	if (channel->type == IIO_ACCEL) {
+		iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+
+		/* The device has to be in standby mode in order to be able to
+		 * modify the offset.
+		 * Obtain the current op mode */
+		op_mode = adxl313->op_mode;
+		/* Set the device in standby mode */
+		ret = adxl313_set_op_mode(adxl313, ADXL313_STDBY);
+		if (ret)
+			return ret;
+
+		/* Set the offset on the requested axis */
+		switch (channel->address) {
+		case ACCEL_AXIS_X:
+			ret = adxl313_set_offset(adxl313, val, ADXL313_X_AXIS);
+			break;
+		case ACCEL_AXIS_Y:
+			ret = adxl313_set_offset(adxl313, val, ADXL313_Y_AXIS);
+			break;
+		case ACCEL_AXIS_Z:
+			ret = adxl313_set_offset(adxl313, val, ADXL313_Z_AXIS);
+			break;
+		default:
+			ret = -EINVAL;
+			break;
+		}
+
+		if (ret) {
+			adxl313_set_op_mode(adxl313, op_mode);
+			return ret;
+		}
+
+		return adxl313_set_op_mode(adxl313, op_mode);
+	}
+
+	return -EINVAL;
+}
+
+/*******************************************************************************
+ * @brief Handles the read request for sampling_frequency attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl313_iio_read_sampling_freq(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2] = { 0 };
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	if (channel->type == IIO_ACCEL) {
+		vals[0] = adxl313_iio_odr_table[adxl313->odr][0];
+		vals[1] = adxl313_iio_odr_table[adxl313->odr][1];
+		return iio_format_value(buf, len, IIO_VAL_INT_PLUS_MICRO, 1, vals);
```

Comment:
Shouldn't this  `len` be 2?

Reply:
Yest, this is the correct version, although it works this way (work with array).